### PR TITLE
Spike: auto detection test

### PR DESCRIPTION
**Pull request by lili2311**
- [ ] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team
WIP

### CODE REVIEW

File: src/lib/snyk-test/find-globs.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { getAllSupportedManifestFiles } from './project-types';
+
+
+export function fileExists(root, file) {
+  return file && fs.existsSync(root) &&
+    fs.existsSync(pathLib.resolve(root, file));
+}
+
+export function findGlobs(searchPath: string): string[] {
+  const manifestFiles = getAllSupportedManifestFiles();
```
Comment (lili2311) [inline src/lib/snyk-test/find-globs.ts]: change this to get default manifest only

File: src/lib/snyk-test/project-types.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+const PROJECT_TYPES_WITH_MANIFEST: {
+  readonly [packageManager in SupportedProjectTypesWithManifest]: PackageManagersConfig;
+} = {
+  npm: {
+    manifestFiles: ['package.json', 'package-lock.json'],
+    defaultManifestFile: 'package.json',
+  },
+  rubygems: {
```
Comment (lili2311) [inline src/lib/snyk-test/project-types.ts]: only `ruby` & `npm` for now

File: src/lib/snyk-test/index.js [hunk 3]
```diff
@@ -5,17 +5,23 @@ const runTest = require('./run-test');
...
 function test(root, options, callback) {
+
   if (typeof options === 'function') {
     callback = options;
     options = {};
   }
   if (!options) {
     options = {};
   }
+  const searchPath = getSearchPath();
```
Comment (lili2311) [inline src/lib/snyk-test/index.js]: what if we have `--folder` provided, what if we have `--file` => not supported, throw error?

File: src/cli/commands/test/index.ts [hunk 4]
```diff
@@ -468,7 +467,7 @@ function rightPadWithSpaces(s, desiredLength) {
 
 function metaForDisplay(res, options) {
   const padToLength = 19; // chars to align
-  const packageManager = options.packageManager || res.packageManager;
+  const packageManager = res.packageManager || options.packageManager ;
```
Comment (sfat) [inline src/cli/commands/test/index.ts]: whitespace before `;`

**Pull request closed without merging**